modules/Vtiger/ExecuteFunctions.php

```diff
@@ -693,6 +693,31 @@
 			);
 		}
 		break;
+	case 'getDeepFieldValue':
+		$moduleName = vtlib_purify($_REQUEST['moduleName']);
+		$value = vtlib_purify($_REQUEST['value']);
+		$fieldsArray = explode("-&gt;", $value);
+		// remove all $ signs
+		foreach ($fieldsArray as &$value) {
+			$value = substr($value, 1);
+		}
+		$firstFieldRecordID = vtlib_purify($_REQUEST['firstFieldRecordID']);
+
+		$result = '';
+		$currentFieldvalue = $firstFieldRecordID;
+		foreach ($fieldsArray as $key => $fieldName) {
+			if ($key == 0) {
+				continue;
+			}
+			$queryGenerator = new QueryGenerator(getSalesEntityType($currentFieldvalue), $current_user);
```

Comment:
this is expensive, calling the class and launching a query inside the loop for each field when we can do one query  and the whole function overlaps a lot with `getFieldValuesFromRecord`, look for me and we discuss